zebra-state/src/config.rs

```diff
@@ -512,10 +510,9 @@ pub(crate) mod hidden {
 
         let version = format!("{}.{}", changed_version.minor, changed_version.patch);
 
-        // # Concurrency
-        //
-        // The caller handles locking for this file write.
-        fs::write(version_path, version.as_bytes())?;
+        // Write to a temporary file, so the cache is not corrupted if Zebra shuts down or crashes
+        // at the same time.
```

Comment:
###### Optional Suggestion

```suggestion
        // Write the version file atomically so the cache is not corrupted if Zebra shuts down or
        // crashes.
```